bluemira/power_cycle/net.py

```diff
@@ -0,0 +1,660 @@
+            )
+
+    @property
+    def duration(self):
+        """Duration of phase"""
+        data = [s_ph.duration for s_ph in self.subphases.values()]
+        try:
+            return getattr(np, self.config.operation)(data)
+        except np.core._exceptions.UFuncTypeError as e:
+            raise TypeError(f"duration variables have not been imported {data}") from e
+
+
+@dataclass
+class PowerCycleLibraryConfig:
+    """Power Cycle Configuration"""
+
+    scenario: ScenarioConfigDescriptor = ScenarioConfigDescriptor()
+    pulse: LibraryConfigDescriptor = LibraryConfigDescriptor(config=PulseConfig)
+    phase: LibraryConfigDescriptor = LibraryConfigDescriptor(config=PhaseConfig)
+    subphase: LibraryConfigDescriptor = LibraryConfigDescriptor(
+        config=PowerCycleSubPhase
+    )
+    system: LibraryConfigDescriptor = LibraryConfigDescriptor(config=PowerCycleSystem)
+    subsystem: LibraryConfigDescriptor = LibraryConfigDescriptor(
+        config=PowerCycleSubSystem
+    )
+    loads: LibraryConfigDescriptor = LibraryConfigDescriptor(config=PowerCycleLoadConfig)
+
+    def check_config(self):
+        """Check powercycle configuration"""
+        ph_keys = self.phase.keys()
+        sph_keys = self.subphase.keys()
+        ss_keys = self.subsystem.keys()
+        loads = self.loads.keys()
+        # scenario has known pulses
+        if unknown_pulse := self.scenario.pulses.keys() - self.pulse.keys():
+            raise ValueError(f"Unknown pulses {unknown_pulse}")
+        # pulses have known phases
+        for pulse in self.pulse.values():
+            if unknown_phase := pulse.phases - ph_keys:
+                raise ValueError(f"Unknown phases {unknown_phase}")
+        # phases have known subphases
+        for ph_c in self.phase.values():
+            if unknown_s_ph := ph_c.subphases - sph_keys:
+                raise ValueError(f"Unknown subphase configurations {unknown_s_ph}")
+        # subphases have known loads
+        for subphase in self.subphase.values():
+            if unknown_load := subphase.loads - loads:
+                raise ValueError(f"Unknown loads {unknown_load}")
+
+        # systems have known subsystems
+        for sys_c in self.system.values():
+            if unknown := sys_c.subsystems - ss_keys:
+                raise ValueError(f"Unknown subsystem configurations {unknown}")
+        # subsystems have known loads
+        for s_sys_c in self.subsystem.values():
+            if unknown_load := s_sys_c.loads - loads:
+                raise ValueError(f"Unknown loads {unknown_load}")
+
+    def import_subphase_duration(self, subphase_duration_params):
+        """Import subphase data"""
+        for s_ph in self.subphase.values():
+            if isinstance(s_ph.duration, str):
+                s_ph.duration = getattr(
+                    subphase_duration_params, s_ph.duration.replace("$", "")
+                )
+
```

Comment:
```suggestion
    def set_subphase_durations(self, duration_values: Dict[str, float]):
        """Import subphase data"""
        for s_ph in self.subphase.values():
            if isinstance(s_ph.duration, str):
                # force the lookup as it should fail if not found
                # i.e. all durations should be imported
                s_ph.duration = duration_values[s_ph.duration.replace("$", "")]
```

This could be made private and called during the __init__ where duration_values could be optionally provided.

Could make it ensure that all str durations are set with values during the init of the config